### PR TITLE
fix(config): reject an explicit empty --sql value

### LINKED ISSUE #660: [BUG] reject an explicit empty --sql value
## Summary
An explicit empty `--sql` value is silently treated as if `--sql` was never provided.

## Reproduction
```bash
sqly --sql '' testdata/user.csv
```

## Actual behavior
- The command exits `0`.
- Nothing is printed.
- No query is executed.

## Expected behavior
- An explicit empty `--sql` should fail fast with a clear validation error such as `--sql cannot be empty`.

## Impact
- CI jobs and scripts can report success while executing no SQL at all.
- This is easy to trigger from shells, wrappers, or templated commands that expand to an empty string.
- The behavior is inconsistent with other flags such as `--sql-file`, `--output`, `--save-dir`, and `--stdin`, which already reject explicit empty values.

## Suggested fix
- Reject `--sql ""` during argument validation, the same way other string flags are validated when `flag.Changed(...)` is true.

## Suggested regression coverage
- Add an argument-parsing test that passes `--sql ""` and expects a validation error.
- Add a CLI/spec test that asserts `sqly --sql '' testdata/user.csv` exits non-zero and prints a clear message.

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

An explicit empty `--sql ""` was silently treated as if `--sql` was never provided: the command exited `0` without printing anything or running any query. That let CI jobs, wrappers, and templated commands that expand to an empty string report success while executing no SQL.

This validates `--sql` during argument parsing the same way `--output`, `--sql-file`, `--save-dir`, and `--stdin` are already validated when their flag is changed, returning a clear `--sql requires a non-empty SQL statement` error so the malformed value is visible.

## Changes

- `config/argument.go`: reject `--sql ""` when the flag is explicitly set to an empty string.
- `config/errors.go`: add `errEmptyQuery`.
- `config/argument_test.go`: add a unit test asserting the empty-`--sql` rejection.
- `spec/v0_25_bugs_spec.sh`: add a binary-level shellspec case asserting `sqly --sql '' testdata/user.csv` exits non-zero with a `--sql` hint.

## Test

- `go test ./config/`
- `shellspec --shell sh spec/v0_25_bugs_spec.sh`

Closes #660

## Summary by CodeRabbit

* **Bug Fixes**
  * `--sql ""` is now rejected with a clear validation error instead of being treated as a successful no-op.
  * Improved validation behavior when `--sql` is explicitly provided as an empty string.
  * Added regression coverage to ensure the command fails for empty `--sql` values and emits the expected message.

* **Chores**
  * Updated the changelog with the new bug fix entry.

### CODE REVIEW

File: spec/v0_25_bugs_spec.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  BeforeEach 'setup'
+  AfterEach 'cleanup'
+
+  # An explicit empty --sql value must fail fast instead of silently running no
+  # query and exiting 0, matching the other string flags that reject empty values.
+  It 'rejects an explicit empty --sql value'
+    When run sqly --sql '' "$PROJECT_ROOT/testdata/user.csv"
+    The status should be failure
+    The stderr should include '--sql'
```
[the remark this answers is not in the document]
Comment (nao1215) [reply spec/v0_25_bugs_spec.sh]: Good catch. Tightened the spec to assert the full message "--sql requires a non-empty SQL statement" so an unrelated stderr mentioning the flag cannot pass.
